File: framework/languages/java/samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/play/adapters/zlink/spots/handlers/PlayerWinMilestoneEventHandler.kt
package systems.zlink.samples.kotlin.tictactoe.server.play.adapters.zlink.spots.handlers

import systems.zlink.framework.spots.ZLinkSpotSubscriptionHandler
import systems.zlink.samples.kotlin.tictactoe.server.play.adapters.zlink.spots.PlayEntrySpot
import systems.zlink.samples.kotlin.tictactoe.shared.contracts.PlayerWinMilestoneEvent

class PlayerWinMilestoneEventHandler :
    ZLinkSpotSubscriptionHandler<PlayEntrySpot, PlayerWinMilestoneEvent> {
    override fun handle(spot: PlayEntrySpot, message: PlayerWinMilestoneEvent) {
        spot.notifyMilestone(message)
    }
}

File: framework/languages/node/samples/TicTacToe.Ts/Server/Play/Infrastructure/ZLink/Spots/EntrySpot/Handlers/player-win-milestone-event-handler.ts
import { Injectable } from '@nestjs/common';
import type { ZLinkPublishContext, ZLinkSpotSubscriptionHandler } from '@zlink-systems/framework';
import type { PlayerWinMilestoneMsg } from '../../../../../../../Shared/Contracts/messages';
import type { PlayEntrySpot } from '../play-entry-spot';

@Injectable()
class PlayerWinMilestoneEventHandler
  implements ZLinkSpotSubscriptionHandler<PlayEntrySpot, PlayerWinMilestoneMsg> {
  async handle(
    entrySpot: PlayEntrySpot,
    event: PlayerWinMilestoneMsg,
    context: ZLinkPublishContext
  ): Promise<void> {
    void context;
    await entrySpot.notifyMilestone(event);
  }
}

export { PlayerWinMilestoneEventHandler };
